desc/magnetic_fields/_current_potential.py

```diff
@@ -644,7 +647,23 @@ def _compute_magnetic_field_from_CurrentPotentialField(
     # compute surface current, and store grid quantities
     # needed for integration in class
     # TODO: does this have to be xyz, or can it be computed in rpz as well?
-    data = field.compute(["K", "x"], grid=source_grid, basis="xyz", params=params)
+    if not params and not transforms:
```

Comment:
why this conditional?

Reply:
To avoid calling `field.compute` inside a compute function of an objective, as there is some logic there that `jit` does not like